File: apps/api_server/src/models/gmail_signal.ts
export interface GmailSignal {
  id: string;
  ownerId: number | null;
  externalId: string;
  threadId: string;
  fromName: string | null;
  fromEmail: string | null;
  subject: string | null;
  snippet: string | null;
  receivedAt: string | null;
  isUnread: boolean;
  createdAt: string;
  updatedAt: string;
}

File: apps/desktop_flutter/lib/features/agent_email/models/gmail_signal.dart
class AgentEmailGmailSignal {
  AgentEmailGmailSignal({
    required this.id,
    required this.fromName,
    required this.fromEmail,
    required this.subject,
    this.snippet,
    this.receivedAt,
    required this.isUnread,
  });

  factory AgentEmailGmailSignal.fromJson(Map<String, dynamic> json) {
    return AgentEmailGmailSignal(
      id: json['id'] as String? ?? json['externalId'] as String? ?? '',
      fromName: json['fromName'] as String? ?? '',
      fromEmail: json['fromEmail'] as String? ?? '',
      subject: json['subject'] as String? ?? '(No subject)',
      snippet: json['snippet'] as String?,
      receivedAt: json['receivedAt'] as String?,
      isUnread: json['isUnread'] as bool? ?? false,
    );
  }

  final String id;
  final String fromName;
  final String fromEmail;
  final String? snippet;
  final String? receivedAt;
  final bool isUnread;
  final String subject;
}
